search-app/src/cmr/search/api/concepts_search.clj

```diff
@@ -0,0 +1,223 @@
+
+      :else
+      {:status 415
+       :headers {common-routes/CORS_ORIGIN_HEADER "*"}
+       :body (str "Unsupported content type ["
+                  (get headers (string/lower-case common-routes/CONTENT_TYPE_HEADER)) "]")})))
+
+(defn- get-granules-timeline
+  "Retrieves a timeline of granules within each collection found."
+  [ctx path-w-extension params headers query-string]
+  (let [params (process-params :granule params path-w-extension headers mt/json)
+        _ (info (format "Getting granule timeline from client %s with params %s."
+                        (:client-id ctx) (pr-str params)))
+        search-params (lp/process-legacy-psa params)
+        results (query-svc/get-granule-timeline ctx search-params)]
+    {:status 200
+     :headers {common-routes/CORS_ORIGIN_HEADER "*"}
+     :body results}))
+
+(defn- find-concepts-by-aql
+  "Invokes query service to parse the AQL query, find results and returns the response"
+  [ctx path-w-extension params headers aql]
+  (let [params (process-params nil params path-w-extension headers mt/xml)
+        _ (info (format "Searching for concepts from client %s in format %s with AQL: %s and query parameters %s."
+                        (:client-id ctx) (rfh/printable-result-format (:result-format params)) aql params))
+        results (query-svc/find-concepts-by-aql ctx params aql)]
+    (core-api/search-response ctx results)))
+
+(defn- find-tiles
+  "Retrieves all the tiles which intersect the input geometry"
+  [ctx params]
+  (let [results (query-svc/find-tiles-by-geometry ctx params)]
+    {:status 200
+     :headers {common-routes/CONTENT_TYPE_HEADER (mt/with-utf-8 mt/json)}
+     :body results}))
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;;; Route Definitions
+
+(defn- get-deleted-collections
+  "Invokes query service to search for collections that are deleted and returns the response"
+  [ctx path-w-extension params headers]
+  (let [params (process-params :collection params path-w-extension headers mt/xml)]
+    (info (format "Searching for deleted collections from client %s in format %s with params %s."
+                  (:client-id ctx) (rfh/printable-result-format (:result-format params))
+                  (pr-str params)))
+    (core-api/search-response ctx (query-svc/get-deleted-collections ctx params))))
+
+(defn- get-deleted-granules
+  "Invokes query service to search for granules that are deleted and returns the response"
+  [ctx path-w-extension params headers]
+  (let [params (process-params nil params path-w-extension headers mt/xml)]
+    (info (format "Searching for deleted granules from client %s in format %s with params %s."
+                  (:client-id ctx) (rfh/printable-result-format (:result-format params))
+                  (pr-str params)))
+    (core-api/search-response ctx (query-svc/get-deleted-granules ctx params))))
+
+(def search-routes
```

Comment:
Several of the vars below here need docstrings.

Reply:
Added.